src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int TOTAL_FIGURE_TYPES = 5;
+    private static final int FIGURE_SIDE_MAX_SIZE = 55;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        int figureTypeIndex = random.nextInt(TOTAL_FIGURE_TYPES);
+        switch (figureTypeIndex) {
+            case 0:
+                Square square = new Square();
+                square.setColor(colorSupplier.getRandomColor());
+                square.setSide(random.nextInt(FIGURE_SIDE_MAX_SIZE));
+                figure = square;
+                break;
+            case 1:
+                Rectangle rectangle = new Rectangle();
+                rectangle.setColor(colorSupplier.getRandomColor());
+                rectangle.setSideA(random.nextInt(FIGURE_SIDE_MAX_SIZE));
+                rectangle.setSideB(random.nextInt(FIGURE_SIDE_MAX_SIZE));
+                figure = rectangle;
+                break;
```

Comment:
```suggestion
                return getRandomRetangle();
```
lets create a private method to get each random figure - this will make your code easier to read